Orange/widgets/unsupervised/tests/test_owcorrespondence.py

```diff
@@ -51,3 +51,20 @@ def test_data_one_value_zero(self):
             [(0,), (0,), (0,)]
         )
         self.send_signal(self.widget.Inputs.data, table)
+
+    def test_no_discrete_variables(self):
+        """
+        Do not crash when there are no discrete (categorical) variable(s).
+        GH-2723
+        """
+        table = Table(
+            Domain(
+                [ContinuousVariable("a")]
+            ),
+            [(1,), (2,), (3,)]
+        )
+        self.assertFalse(self.widget.Error.no_disc_vars.is_shown())
+        self.send_signal(self.widget.Inputs.data, table)
+        self.assertTrue(self.widget.Error.no_disc_vars.is_shown())
+        self.send_signal(self.widget.Inputs.data, None)
+        self.assertFalse(self.widget.Error.no_disc_vars.is_shown())
```

Comment:
Add

```
        self.send_signal(self.widget.Inputs.data, table)
        self.assertTrue(self.widget.Error.no_disc_vars.is_shown())
        self.send_signal(self.widget.Inputs.data, Table("iris"))
        self.assertFalse(self.widget.Error.no_disc_vars.is_shown())
```

We have had bugs in the past where the error message was sometimes not cleared.